x-pack/plugins/triggers_actions_ui/public/application/sections/actions_connectors_list/components/actions_connectors_list.tsx

```diff
@@ -127,7 +132,13 @@ export const ActionsConnectorsList: React.FunctionComponent = () => {
 
   async function editItem(connectorTableItem: ActionConnectorTableItem) {
     setEditedConnectorItem(connectorTableItem);
-    setEditFlyoutVisibility(true);
+    if (connectorTableItem.isPreconfigured) {
+      setPreconfiguredFlyoutVisibility(true);
+      setEditFlyoutVisibility(false);
+    } else {
+      setEditFlyoutVisibility(true);
+      setPreconfiguredFlyoutVisibility(false);
+    }
```

Comment:
Looking at this it occurs to me that we would never want to show both the PreConfigured flyout and the Edit flyout at the same time, so maybe we should have only one flyout here?
We can do something like have a single component that decides which flyout to render, for example, something like:

```
const EditConnectorFlyout = (connectorTableItem) => connectorTableItem. isPreconfigured ? < PreconfiguredConnectorFlyout /> : < ConnectorEditFlyout />
```

What that would allow us to do is have only one flyout in the table component and that will prevent us from ever accidentally trying to show both `ConnectorEditFlyout` and `PreconfiguredConnectorFlyout` at the same time. It woiuld also mean having only one `setState` for whether we're displaying a flyout or not.

This feels more maintainable for the future, too. :)

Reply:
@gmmorris with the changes you merged for the TLS / encryption key messages, could this fall as another message to display in the `ConnectorEditFlyout`? (could re-use the same flyout and add a "pre-configured" beta bage in the header)